src/lib/logs/log_forwarder.py

```diff
@@ -132,6 +132,9 @@ def perform_flush(worker_state: WorkerState,
                 context.self_monitoring.sent_logs_entries += len(worker_state.jobs)
                 context.self_monitoring.log_ingest_payload_size += display_payload_size
                 send_batched_acks(subscriber_client, subscription_path, worker_state.ack_ids)
+        elif worker_state.ack_ids:
+            # Send ACKs if processing all messages has failed
```

Comment:
Will it ACK if messages were correctly processed but Dynatrace AG endpoint is not available?

Reply:
looking at the code - I think it will not, wchich is correct :)

Reply:
You're right- if any message will be correctly processed, then interpreter will follow execution branch on line 122, and ACK only if logs have been correctly sent to Dynatrace.